src/main/java/com/exactpro/th2/act/rules/FieldCheckRule.java

```diff
@@ -66,8 +67,21 @@ protected boolean checkMessageFromConnection(Message message) {
     }
 
     private boolean checkExpectedField(MessageOrBuilder message, String fieldName) {
-        Value value = message.getFieldsMap().get(fieldName);
-        return value != null
-                && Objects.equals(expectedFieldValue, value.getSimpleValue());
+        Value actualFieldValue = message.getFieldsMap().get(fieldName);
+        if (actualFieldValue != null){
+            return Objects.equals(expectedFieldValue, actualFieldValue.getSimpleValue());
+        }
+        AtomicBoolean result = new AtomicBoolean(false);
+        message.getFieldsMap().forEach((key, value) -> {
+            if (value.hasMessageValue()) {
+                result.set(result.get() || checkExpectedField(value.getMessageValue(), fieldName));
+            }
+            if (value.hasListValue()) {
+                for (Value listEntry : value.getListValue().getValuesList()) {
+                    result.set(result.get() || checkExpectedField(listEntry.getMessageValue(), fieldName));
+                }
+            }
+        });
+    return result.get();
```

Comment:
Probably it will be better to replace the forEach call with foreach loop
```suggestion
        for (Value value : message.getFieldsMap().values()) {
            if (value.hasMessageValue()) {
                if (checkExpectedField(value.getMessageValue(), fieldName)) {
                    return true;
                }
            }
            if (value.hasListValue()) {
                for (Value listEntry : value.getListValue().getValuesList()) {
                    if (listEntry.hasMessageValue()) {
                        if (checkExpectedField(listEntry.getMessageValue(), fieldName)) {
                            return true;
                        }
                    }
                }
            }
        });
        return false;
```